### ISSUE TITLE
AndoirdJNYHelper.GetSignature: using Byte parameters is obsolete, use SByte parameters instead.

### ISSUE THREAD

**Issue by elritualk** [playgameservices/play-games-plugin-for-unity#3046]
Hi, i'm using Play Games Plugin version 0.10.12 on Unity version 2020.3.2f1.
I use cloud load / save feature, and I have this warnings everytime I save or load a game from cloud.

```
AndroidJNYHelper.GetSignature: using Byte parameters is obsolete, use SByte parameters instead.
AndroidJNIHelper: converting Byte array is obsolete, use SByte array instead.
```

This messages come after I call:

```
byte[] data = PlayerManager.Instance.profile.ToBytes();
((PlayGamesPlatform)Social.Active).SavedGame.CommitUpdate(game, updatedMetadata, data, SavedGameWritten);
```

That calls AndroidSavedGameClient.cs from your plugin:

```
convertedMetadata.JavaContents.Call<bool>("writeBytes", updatedBinaryData)
```

When GPGS call this function is when the warnings appears.

For most players, including me, the cloud load / save feature works good anyway. But I got some other players that when try to load a saved game from cloud, they got a 0 bytes of data from:

```
((PlayGamesPlatform)Social.Active).SavedGame.ReadBinaryData(game, SavedGameLoaded);
public void SavedGameLoaded(SavedGameRequestStatus status, byte[] data) { ... }
```
What can I do? Thanks.

**Comment by neetish-borkar** [comment]
It is just a warning. I am getting it too, but my `Encoding.ASCII.GetBytes()` and `Encoding.ASCII.GetString()` works fine.

**Comment by elritualk** [comment]
> It is just a warning. I am getting it too, but my `Encoding.ASCII.GetBytes()` and `Encoding.ASCII.GetString()` works fine.

Yeah, for most of my players it is just a warning. But for a small part of them its a trouble and I think it may be related with this.

**Comment by toxikman** [comment]
@ozdemir08 Hi Mehmet, account to github search, there have been 3 issues submitted regarding this logspam warning in Unity 2020. It's been almost 8 months. What's the status on getting this fixed?

**Comment by kyleyiso** [comment]
same promblem here,and i dont konw how to fix it !!

**Comment by KyrXtz** [comment]
> same promblem here,and i dont konw how to fix it !!
There is nothing to fix on your side, they have to change the parameter of the called method.
The most you can do is hide the warning.

**Comment by Goisak100** [comment]
@KyrXtz Hello, Hello, I also have the same problem. I think my guess is that it's because `CommitUpdate()` and `ReadBinaryData()` use a `byte[]` parameter.
I don't think there is an API that uses `sbyte[]` instead of `byte[]`. Is there any way to modify that parameter?
Or can you tell me the proper way to hide that warning?

**Comment by toxikman** [comment]
> Nearly two years later this issue is still present (occurs currently in latest release [0.11.01](https://github.com/playgameservices/play-games-plugin-for-unity/releases/tag/v0.11.01). When can we expect this to be resolved?

I suspect this Unity plug-in is now abandonware. 😞  The documentation copyright still says 2014. Probably best to just make the conversion modifications ourselves. Honestly I'm surprised nobody has submitted a pull request yet.

**Comment by JVimes** [comment]
I can't find `GetSignature` or `AndoirdJNYHelper` in code. Maybe the error originated on server?

**Comment by vaibhavgpt-google** [comment]
Closing this issue as part of a cleanup of older tickets. Google Play Games v1 is now deprecated. Please upgrade to version v2.1.0 or newer: https://github.com/playgameservices/play-games-plugin-for-unity/releases/tag/v2.1.0. If this issue persists on the new v2 architecture, please open a new ticket.
